pkg/solana/txm/pendingtx.go

```diff
@@ -138,28 +210,283 @@ func (c *pendingTxContext) ListAll() []solana.Signature {
 }
 
 // Expired returns if the timeout for trying to confirm a signature has been reached
-func (c *pendingTxContext) Expired(sig solana.Signature, lifespan time.Duration) bool {
+func (c *pendingTxContext) Expired(sig solana.Signature, confirmationTimeout time.Duration) bool {
```

Comment:
should we use the new lock methods for consistency here?

Reply:
I had considered this but decided to only use the new methods for just the state transition methods. They have similar returns like id and error which some of these other methods don't. I could rework it to set fields outside of the `wthReadLock` method but thought it was easier to leave the methods that followed the normal locking standard.
```
c.lock.Lock()
defer c.lock.Unlock()
```